### ISSUE TITLE
Host Rename. LXD Fails to start

### ISSUE THREAD

**Issue by sidxz** [lxc/lxd#5086]
Is there is a way to update lxd (likely db) after the hostname of the physical server is changed?
It seems lxd is querying dns with the older hostname which fails as in dns no corresponding entry exists.

lxd version 3.0.0

**Comment by stgraber** [comment]
LXD shouldn't care about the name of your physical system at all, what exactly is failing?

**Comment by stgraber** [comment]
The one thing I can think of is if you used your system's name in core.https_address maybe, in which case your easiest way out is to add a temporary entry to /etc/hosts, start LXD and set core.https_address to another value, quite possibly an IP address to avoid this kind of issue.

**Comment by sidxz** [comment]
Hi stgraber,
Thank you, It is related to lxd cluster. since the hostname is changed, the cluster url doesn't work anymore

**Comment by sidxz** [comment]
Is there is a way to change the cluster url?

**Comment by stgraber** [comment]
What cluster URL?

**Comment by sidxz** [comment]
lxc cluster list
>               NAME                           |                 URL                  | DATABASE | STATE  |      MESSAGE

**Comment by stgraber** [comment]
Please paste the output of:
```
lxd sql global "SELECT * FROM nodes;"
lxd sql local "SELECT * FROM raft_nodes;"
lxd sql local "SELECT * FROM config;"
```

**Comment by stgraber** [comment]
Node address or in your case FQDN are normally completely locked and can't be changed.
If your database is online, I can give you commands that will directly alter those records but I can't guarantee that the database will like that very much and you may need to go fix the local database of all your nodes one by one afterwards.

**Comment by sidxz** [comment]
This is a dev env. 
I updated values directly in sqlite3. lxd fails to start now 'Error: failed to open cluster database: failed to ensure schema: failed to update node version info: updated 0 rows instead of 1'

lxd --verbose
[Leader] entering Leader state 
EROR[09-28|17:04:13] Failed to start the daemon: failed to open cluster database: failed to ensure schema: failed to update node version info: updated 0 rows instead of 1 

I realize that there had been major changes in the next release regarding clusters. Let me try this in an updated version.

**Comment by stgraber** [comment]
What database did you update exactly?

You'd have needed to update the global database first, which is RAFT and can't be updated directly through sqlite3, then you'd have needed to update the local database of all nodes (which is sqlite3) and then restart LXD on all nodes.

Altering the local database or the temporary binary version of the global database may or may not have completely messed up your replicated state...

But yes, 3.0.0 is rather ancient as far as dqlite goes, 3.0.1 at least has the tools to query the raft database directly (the `lxd sql` command) and 3.0.2 would be using the newer dqlite implementation which fixes a few rough edges.

In general for clusters, the node address is something you should never change, so you should set it to whatever is least likely to change between an IP or DNS name.

**Comment by GuanglinDu** [comment]
Dear @stgraber , I ran across this problem after I changed the hostname of the cluster members. LXD didn't follow suit. I had to use the original cluster member name when running LXC commands with flag --target, e.g.,
`lxc copy archlinux1 archlinux2 --target old_hostname`
As you mentioned above, can I update the LXD db manually this way?
`lxd sql global "UPDATE nodes SET name="newname" WHERE name="oldname:;"`
It seems there'll be not harm to the whole LXD/LXC system.
